pkg/table/contextimpl/table.go

```diff
@@ -55,9 +55,24 @@ func (ctx *TableContextImpl) GetExprCtx() exprctx.ExprContext {
 	return ctx.Context.GetExprCtx()
 }
 
+// ConnectionID implements the MutateContext interface.
+func (ctx *TableContextImpl) ConnectionID() uint64 {
+	return ctx.vars().ConnectionID
+}
+
 // InRestrictedSQL returns whether the current context is used in restricted SQL.
 func (ctx *TableContextImpl) InRestrictedSQL() bool {
-	return ctx.vars().StmtCtx.InRestrictedSQL
```

Comment:
I found `StmtCtx` and `SessionVars` all have `InRestrictedSQL` and they are kept the same in `ResetContextOfStmt`:

https://github.com/pingcap/tidb/blob/2da324d1a1831c7be31ef17a164ff511757cdbb5/pkg/executor/executor.go#L1946

Read it from `SessionVars` instead of `StmtCtx` keeps the same behavior as previous implementations.